File: android_claude/data/crypto/src/main/java/com/zarabandajose/transcripto/data/crypto/repository/CipherRepositoryImpl.kt
package com.zarabandajose.transcripto.data.crypto.repository

import com.zarabandajose.transcripto.core.common.Result
import com.zarabandajose.transcripto.data.crypto.cipher.Base64Codec
import com.zarabandajose.transcripto.data.crypto.cipher.CaesarCipher
import com.zarabandajose.transcripto.data.crypto.cipher.VigenereCipher
import com.zarabandajose.transcripto.data.crypto.cipher.XorCipher
import com.zarabandajose.transcripto.domain.model.CipherMethod
import com.zarabandajose.transcripto.domain.model.CipherParams
import com.zarabandajose.transcripto.domain.repository.CipherRepository
import javax.inject.Inject

/**
 * Implementación del repositorio de cifrado.
 * Delega las operaciones a los diferentes cifradores según el método seleccionado.
 */
class CipherRepositoryImpl @Inject constructor(
    private val caesarCipher: CaesarCipher,
    private val base64Codec: Base64Codec,
    private val vigenereCipher: VigenereCipher,
    private val xorCipher: XorCipher
) : CipherRepository {
    
    override suspend fun encrypt(input: String, params: CipherParams): Result<String> {
        return when (params.method) {
            CipherMethod.CAESAR -> caesarCipher.encrypt(input, params)
            CipherMethod.BASE64 -> base64Codec.encrypt(input, params)
            CipherMethod.VIGENERE -> vigenereCipher.encrypt(input, params)
            CipherMethod.XOR -> xorCipher.encrypt(input, params)
        }
    }
    
    override suspend fun decrypt(input: String, params: CipherParams): Result<String> {
        return when (params.method) {
            CipherMethod.CAESAR -> caesarCipher.decrypt(input, params)
            CipherMethod.BASE64 -> base64Codec.decrypt(input, params)
            CipherMethod.VIGENERE -> vigenereCipher.decrypt(input, params)
            CipherMethod.XOR -> xorCipher.decrypt(input, params)
        }
    }
}

File: android_flutter_claude/transcripto/packages/data_crypto/lib/src/repositories/cipher_repository_impl.dart
import 'package:core_common/core_common.dart';
import 'package:domain/domain.dart';
import '../ciphers/caesar_cipher.dart';
import '../ciphers/base64_codec_impl.dart';
import '../ciphers/vigenere_cipher.dart';
import '../ciphers/xor_cipher.dart';

/// Implementation of cipher repository
class CipherRepositoryImpl implements CipherRepository {
  const CipherRepositoryImpl();
  
  static const _caesarCipher = CaesarCipher();
  static const _base64Codec = Base64CodecImpl();
  static const _vigenereCipher = VigenereCipher();
  static const _xorCipher = XorCipher();
  
  @override
  Result<String> encrypt(String input, CipherParams params) {
    try {
      final result = switch (params.method) {
        CipherMethod.caesar => _caesarCipher.encrypt(input, params),
        CipherMethod.base64 => _base64Codec.encrypt(input, params),
        CipherMethod.vigenere => _vigenereCipher.encrypt(input, params),
        CipherMethod.xor => _xorCipher.encrypt(input, params),
      };
      
      return Success(result);
    } catch (e, stackTrace) {
      return Failure(e, stackTrace);
    }
  }
  
  @override
  Result<String> decrypt(String input, CipherParams params) {
    try {
      final result = switch (params.method) {
        CipherMethod.caesar => _caesarCipher.decrypt(input, params),
        CipherMethod.base64 => _base64Codec.decrypt(input, params),
        CipherMethod.vigenere => _vigenereCipher.decrypt(input, params),
        CipherMethod.xor => _xorCipher.decrypt(input, params),
      };
      
      return Success(result);
    } catch (e, stackTrace) {
      return Failure(e, stackTrace);
    }
  }
}
